pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -941,13 +941,15 @@ ClientConnection::newConsumerStats(uint64_t consumerId, uint64_t requestId) {
 void ClientConnection::newTopicLookup(const std::string& destinationName, bool authoritative,
                                       const uint64_t requestId,
                                       LookupDataResultPromisePtr promise) {
+    Lock lock(mutex_);
```

Comment:
I don't think it's thread safe, we're touching the same object. Especially the baseCmd is going to have different command enum value selected.